oddbird/Metecho
# Allow retrieving datasets from Dev Orgs.

![](https://source.unsplash.com/featured/?cute,animal&220628)

Added a new endpoint at `/api/scratch-orgs/:id/refresh_datasets` that queues a job to parse dataset definitions from the `datasets/` folder of the Task branch. These definitions are cached in the `ScratchOrg.datasets` and are included in all ScratchOrg responses along with any user facing errors in `ScratchOrg.datasets_parse_errors`.

A similar endpoint has been added to refresh schema definitions from a Task Dev org:  `/api/scratch-orgs/:id/refresh_datasets_schema`. The job will  output the schema to `ScratchOrg.dataset_schema`.

Finally, the frontend can also retrieve and commit a dataset by POSTing to `/api/scratch-orgs/:id/commit_dataset`. The request body must include `dataset_name`, `dataset_definition`, and `commit_message`. The request will queue a job that will retrieve the dataset from the Dev org and commit and update definition and data.

# To test

1. Build and run migrations, etc
2. Choose (or create) a Task on `Metecho-Test` project where you have (or create) a Dev Org. Then go to that Task in the admin and set `Branch name` to `feature/epic-with-tasks-16__sample-task-5`
3. Click through to visit the Org in the SF UI, if you haven't already
4. You should see the new "Retrieve Dataset" button above the org cards
